movedata/move.go

```diff
@@ -82,14 +90,24 @@ func (mover *Mover) MoveData() error {
 			return err
 		}
 
+		convertedArrayOfData := casting.CastArrayOfData(columnsMap, cols, columns)
+		for i := range columns {
+			columnPointers[i] = &convertedArrayOfData[i]
```

Comment:
Probably here we can avoid the `columnPointers` map, and write the cast back directly to `document`

Reply:
(consider this as minor)